demosplan/DemosPlanCoreBundle/Logic/Statement/AssessmentTableExporter/AssessmentTableZipExporter.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of the package demosplan.
+ *
+ * (c) 2010-present DEMOS plan GmbH, for more information see the license file.
+ *
+ * All rights reserved
+ */
+
+namespace demosplan\DemosPlanCoreBundle\Logic\Statement\AssessmentTableExporter;
+
+use demosplan\DemosPlanCoreBundle\Entity\File;
+use Exception;
+use PhpOffice\PhpSpreadsheet\Writer\Xlsx;
+
+class AssessmentTableZipExporter extends AssessmentTableXlsExporter
+{
+    protected array $supportedTypes = ['zip'];
+
+    /**
+     * @throws Exception
+     */
+    public function __invoke(array $parameters): array
+    {
+        $xlsxArray = parent::__invoke($parameters);
+
+        $statementAttachments = $this->getAttachmentsOfStatements($parameters['items']);
+
+        /** @var Xlsx $xlsxWriter */
+        $xlsxWriter = $xlsxArray['writer'];
+        $xlsxArray['writer'] = $this->writeReferencesIntoXlsx($xlsxWriter, $statementAttachments);
+
+        return [
+            'zipFileName' => $this->translator->trans('evaluation.assessment.table.export'),
+            'xlsx'        => $xlsxArray,
+            'attachments' => $statementAttachments,
+        ];
+    }
+
+    /**
+     * @return array<int, array<int, File>>
+     *
+     * @throws Exception
+     */
+    private function getAttachmentsOfStatements(array $statementIds): array
+    {
+        $files = [];
+        $index = 0;
+        foreach ($statementIds as $statementId) {
+            $statementAttachments =
+                $this->assessmentHandler->getStatementService()->getFileContainersForStatement($statementId);
```

Comment:
do not fetch a service via another class. Fetch the StatementService by itself

Reply:
I have extended the xlsx exporter so that when I invoke the zip exporter, I can first invoke the xlsx exporter (parent::__invoke()) to use the existing logic of the xlsx exporter and to leave the logic in AssessmentTableExporterStrategy and DemosPlanAssessmentExportController unchanged.